From: Nicholas Marriott <nicholas.marriott@gmail.com>
Origin: http://sourceforge.net/p/tmux/mailman/message/32417093/
Subject: Work around buggy errno handling in cygwin.dll

  Between snapshots 2014-04-18 and 2014-04-24 there were changes made to the
  socket error handling in cygwin1.dll [1] that break tmux [2, 3, 4].

  [1] https://cygwin.com/ml/cygwin/2014-04/msg00539.html
  [2] http://sourceforge.net/p/tmux/mailman/message/32396655/
  [3] http://sourceforge.net/p/tmux/mailman/message/32400834/
  [4] https://cygwin.com/ml/cygwin/2014-06/msg00008.html
diff a/client.c b/client.c
--- a/client.c
+++ b/client.c
@@ -111,7 +111,7 @@ retry:
 		fatal("socket failed");
 
 	if (connect(fd, (struct sockaddr *) &sa, SUN_LEN(&sa)) == -1) {
-		if (errno != ECONNREFUSED && errno != ENOENT)
+		if (errno != ECONNREFUSED && errno != ENOENT && errno != 0)
 			goto failed;
 		if (!start_server)
 			goto failed;
